### PR TITLE
Adapt to PyTorch explicitly representing optionality

### PR DESCRIPTION
**Pull request by smessmer**
PyTorch now explicitly represents optionality of Tensors in operator functions. This changes the signature for the indexing operators since they take lists of optional tensors.

### CODE REVIEW

File: torch_xla/csrc/ops/index_ops.cpp [hunk 1]
```diff
@@ -221,12 +234,18 @@ ir::NodePtr IndexCopyOp(const ir::Value& buffer, xla::int64 dim,
...
   CheckIndexTensorTypes(orig_indices);
   // First expand ByteTensor (boolean masks) into 1 or more LongTensors, then
   // broadcast all index tensors together.
-  auto indices = at::expand_outplace(ExpandByteTensors(base, orig_indices));
+  auto indices_vec = at::expand_outplace(ExpandByteTensors(base, orig_indices));
+  torch::List<c10::optional<at::Tensor>> indices;
+  indices.reserve(indices_vec.size());
+  for (at::Tensor i : std::move(indices_vec)) {
+    indices.push_back(std::move(i));
+  }
```
Comment (JackCaoG) [inline torch_xla/csrc/ops/index_ops.cpp:248]: Why couldn't we just pass `indices_vec`? I didn;t find the function prototype change of `expand_outplace ` in the linked pytorch pr nor can I find the `expand_outplace` that takes `c10::List<c10::optional<at::Tensor>>&` in https://github.com/pytorch/pytorch/blob/71ca600af93b4ad58ca6fc083fa5d42a56b43194/aten/src/ATen/ExpandUtils.h. What am I missing here?

**Review by JackCaoG — CHANGES_REQUESTED** [review]
Mostly LGTM

File: torch_xla/csrc/ops/index_ops.cpp (hunk 1 above)
Comment (smessmer) [reply torch_xla/csrc/ops/index_ops.cpp]: you're right, I changed it

**Review by JackCaoG — APPROVED** [review]
Thanks @smessmer! Feel free to remove the pin and merge it when pytorch pr is merged.

**Comment by smessmer** [comment]
merged in https://github.com/pytorch/xla/commit/3aebd8adf46261bf548e4597c19ffd2577d3d39c

**Pull request closed without merging**